src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIModelMapper.ChatCompletion.cs

```diff
@@ -148,7 +149,12 @@ public static ChatOptions FromOpenAIOptions(OpenAI.Chat.ChatCompletionOptions? o
 
         if (options is not null)
         {
-            result.ModelId = _getModelIdAccessor.Invoke(options, null)?.ToString();
+            result.ModelId = _getModelIdAccessor.Invoke(options, null)?.ToString() switch
+            {
+                null or "" => null,
```

Comment:
Is this sufficient, or is there any situation where there might be non-empty whitespace which should also be null?

Reply:
I thought about it, but in absence of evidence that it occurs I figured I should avoid overcompensating.